src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,118 @@
+package duke;
+
+import duke.Ui.Ui;
+import duke.command.IllegalCommandException;
+import duke.storage.TaskStorage;
+import duke.task.Task;
+import duke.task.TaskDispatch;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class Duke {
+    // attributes
+    private TaskStorage taskStorage;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        try {
+            this.ui = new Ui();
+            this.taskStorage = new TaskStorage(filePath);
+        } catch (FileNotFoundException e) {
+            System.out.println(filePath);
+            System.out.println("File not found");
+        }
+    }
+
+    public static void main(String[] args) {
+        new Duke("./data/Storage.txt").runDuke();
+    }
+
+    // Carry out Add, List, Done commands if entered by user
+    // Terminates when user gives exit signal
+    private void runDuke() {
+        ui.showWelcome();
+
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            String nextInput = scanner.nextLine();
+            // Validation check
+            if (nextInput.equals("bye")) {
+                stopDuke();
+                break;
+            }
+            // Handle List command if any
+            if (nextInput.equals("list")) {
+                handleCommandList();
+                continue;
+            }
+
+            // Check and Handle Done command
+            // Possible source of error
+            String[] commandArgs = nextInput.split(" ");
+
+            if (commandArgs[0].equals("done")) {
+                // duke.task.Todo: Add exception handling
+                handleCommandDone(Integer.parseInt(commandArgs[1]));
+                continue;
+            }
+
+            if (commandArgs[0].equals("delete")) {
+                // duke.task.Todo: Add exception handling
+                handleCommandDelete(Integer.parseInt(commandArgs[1]));
+                continue;
+            }
+
+            // Handle Add Command
+            Task t = null;
+            try {
+                t = TaskDispatch.dispatchTaskFromInput(commandArgs);
+            } catch (IllegalArgumentException e1){
+                System.out.println("    " + e1.getMessage());
+                continue;
+            } catch (IllegalCommandException e2) {
+                System.out.println("    " + e2.getMessage());
+                continue;
+            }
+
+            handleCommandAdd(t, nextInput);
+        }
+    }
+
+    // Print a closing message before stopping duke.Duke
+    private static void stopDuke() {
+        String closingMessage = "Bye. Hope to see you again soon!";
+        System.out.println("    " + closingMessage);
+    }
+
+    // duke.task.Todo: abstract away the following logic from main via new interface for command handlers
+
+    private void handleCommandList() {
```

Comment:
nice! I did it a different way though, my Duke class only has the main method which calls run() on duke. Consider abstracting runDuke() to other handlers!